src/main/java/com/appland/appmap/output/v1/Parameters.java

```diff
@@ -38,58 +40,87 @@ public Parameters() { }
    */
   public Parameters(CtBehavior behavior) {
     MethodInfo methodInfo = behavior.getMethodInfo();
+    String fqn = behavior.getDeclaringClass().getName() +
+      "." + behavior.getName() +
+      methodInfo.getDescriptor();
+
     CodeAttribute codeAttribute = methodInfo.getCodeAttribute();
     if (codeAttribute == null) {
-      throw new NoSourceAvailableException();
+      throw new NoSourceAvailableException("No code attribute for " + fqn);
     }
-
+           
     LocalVariableAttribute locals = (LocalVariableAttribute) codeAttribute.getAttribute(
         javassist.bytecode.LocalVariableAttribute.tag);
 
+    // We should be able to handle methods without debug
+    // information. However, as of 20200822, other errors come up if
+    // do hook them, so bail out here.
     if (locals == null) {
-      throw new NoSourceAvailableException();
+      throw new NoSourceAvailableException("No local variables for " + fqn);
     }
 
-    Integer numberLocals = locals.tableLength();
-    CtClass[] parameterTypes = new CtClass[]{};
-
+    CtClass[] paramTypes = null;
     try {
-      parameterTypes = behavior.getParameterTypes();
+      paramTypes = behavior.getParameterTypes();
     } catch (NotFoundException e) {
-      Logger.println(
-          String.format("failed to get parameter types for %s.%s: %s",
-              behavior.getDeclaringClass().getName(),
-              behavior.getName(),
-              e.getMessage()));
+      throw new NoSourceAvailableException(
+        String.format("Failed to get parameter types for %s: %s",
+                      fqn, e.getMessage()));
     }
 
-    Boolean isStatic = (behavior.getModifiers() & Modifier.STATIC) != 0;
-    Value[] paramValues = new Value[parameterTypes.length];
-    for (int i = 0; i < numberLocals; ++i) {
-      // parameters are not neccesarily the first local variables
-      Integer paramIndex = locals.index(i);
-
-      if (!isStatic) {
-        // index 0 is `this` for nonstatic methods
-        // we don't need it
-        if (paramIndex == 0) {
-          continue;
+    String[] paramNames = null;
+    int numParams = paramTypes.length;
+    if (numParams > 0) {
+      int numLocals = locals.tableLength();
+      
+      // This is handy when debugging this code, put produces too much noise for general use.
+      if (false) {
+        Logger.println("local variables for " + fqn);
```

Comment:
It it worth a system property to activate this?

Reply:
Yup, I'll add one.